### DISCUSSION TITLE
Github deployment getting failed

### DISCUSSION THREAD

**Discussion by Karan-S-dev** [community/community#31351] [Actions]
I am using github actions to deploy my website. CI/CD deployment.

But nowadays its showing a wierd error

"Deployment Failed :(
Deployment Failure Reason: Deployment Canceled"

What is the meaning of this error?

**Comment by airtower-luna** [comment]
Maybe the deployment was canceled due to an error during build. It's impossible to be sure without details though, with Actions problems it's always best if you share the link to an affected workflow.

**Reply by Karan-S-dev** [reply]
Since its a private repo its not possible to share link.
I  have attached the build and deploy description below.

/usr/bin/docker run --name f2bf692f91d24aa1ac032de5ab034a68_789234 --label 229416 --workdir /github/workspace --rm -e "INPUT_AZURE_STATIC_WEB_APPS_API_TOKEN" -e "INPUT_REPO_TOKEN" -e "INPUT_ACTION" -e "INPUT_APP_LOCATION" -e "INPUT_API_LOCATION" -e "INPUT_OUTPUT_LOCATION" -e "INPUT_API_BUILD_COMMAND" -e "INPUT_APP_ARTIFACT_LOCATION" -e "INPUT_APP_BUILD_COMMAND" -e "INPUT_ROUTES_LOCATION" -e "INPUT_SKIP_APP_BUILD" -e "INPUT_CONFIG_FILE_LOCATION" -e "INPUT_SKIP_API_BUILD" -e "INPUT_PRODUCTION_BRANCH" -e "INPUT_DEPLOYMENT_ENVIRONMENT" -e "INPUT_IS_STATIC_EXPORT" -e "HOME" -e "GITHUB_JOB" -e "GITHUB_REF" -e "GITHUB_SHA" -e "GITHUB_REPOSITORY" -e "GITHUB_REPOSITORY_OWNER" -e "GITHUB_RUN_ID" -e "GITHUB_RUN_NUMBER" -e "GITHUB_RETENTION_DAYS" -e "GITHUB_RUN_ATTEMPT" -e "GITHUB_ACTOR" -e "GITHUB_TRIGGERING_ACTOR" -e "GITHUB_WORKFLOW" -e "GITHUB_HEAD_REF" -e "GITHUB_BASE_REF" -e "GITHUB_EVENT_NAME" -e "GITHUB_SERVER_URL" -e "GITHUB_API_URL" -e "GITHUB_GRAPHQL_URL" -e "GITHUB_REF_NAME" -e "GITHUB_REF_PROTECTED" -e "GITHUB_REF_TYPE" -e "GITHUB_WORKSPACE" -e "GITHUB_ACTION" -e "GITHUB_EVENT_PATH" -e "GITHUB_ACTION_REPOSITORY" -e "GITHUB_ACTION_REF" -e "GITHUB_PATH" -e "GITHUB_ENV" -e "GITHUB_STEP_SUMMARY" -e "RUNNER_OS" -e "RUNNER_ARCH" -e "RUNNER_NAME" -e "RUNNER_TOOL_CACHE" -e "RUNNER_TEMP" -e "RUNNER_WORKSPACE" -e "ACTIONS_RUNTIME_URL" -e "ACTIONS_RUNTIME_TOKEN" -e "ACTIONS_CACHE_URL" -e GITHUB_ACTIONS=true -e CI=true -v "/var/run/docker.sock":"/var/run/docker.sock" -v "/home/runner/work/_temp/_github_home":"/github/home" -v "/home/runner/work/_temp/_github_workflow":"/github/workflow" -v "/home/runner/work/_temp/_runner_file_commands":"/github/file_commands" -v "/home/runner/work/Rapidx_Documentation/Rapidx_Documentation":"/github/workspace" 229416:f2bf692f91d24aa1ac032de5ab034a68
DeploymentId: 8b525534-c38e-492d-a2fa-b6a2b55fc191

App Directory Location: '/website' was found.
Looking for event info
Starting to build app with Oryx
Azure Static Web Apps utilizes Oryx to build both static applications and Azure Functions. You can find more details on Oryx here: https://github.com/microsoft/Oryx
---Oryx build logs---


Operation performed by Microsoft Oryx, https://github.com/Microsoft/Oryx
You can report issues at https://github.com/Microsoft/Oryx/issues

Oryx Version: 0.2.20220621.1, Commit: 82ed84f0f3fd3a25ded3029f137601f54e7a1d05, ReleaseTagName: 20220621.1

Build Operation ID: |BwxI/8LgCLU=.58dee2e8_
Repository Commit : fe01a3158f717fe8b0db3559c7ac36506df3eb8e

Detecting platforms...
Detected following platforms:
  nodejs: 17.6.0
Version '17.6.0' of platform 'nodejs' is not installed. Generating script to install it...
Detected the following frameworks: jQuery,React


Source directory     : /github/workspace/website
Destination directory: /bin/staticsites/ss-oryx/app


Downloading and extracting 'nodejs' version '17.6.0' to '/opt/nodejs/17.6.0'...
Downloaded in 0 sec(s).
Verifying checksum...
Extracting contents...
performing sha512 checksum for: nodejs...
Done in 2 sec(s).

Removing existing manifest file
Creating directory for command manifest file if it does not exist
Creating a manifest file...
Node Build Command Manifest file created.

Using Node version:
v17.6.0

Using Npm version:
8.5.1

Running 'npm install --unsafe-perm'...

npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
npm WARN deprecated source-map-url@0.4.1: See https://github.com/lydell/source-map-url#deprecated
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated sntp@1.0.9: This module moved to @hapi/sntp. Please make sure to switch over as this distribution is no longer supported and may contain bugs and critical security issues.
npm WARN deprecated source-map-resolve@0.5.3: See https://github.com/lydell/source-map-resolve#deprecated
npm WARN deprecated gulp-header@1.8.12: Removed event-stream from gulp-header
npm WARN deprecated hoek@2.16.3: This version has been deprecated in accordance with the hapi support policy (hapi.im/support). Please upgrade to the latest version to get the best features, bug fixes, and security patches. If you are unable to upgrade at this time, paid support is available for older versions (hapi.im/commercial).
npm WARN deprecated cryptiles@2.0.5: This version has been deprecated in accordance with the hapi support policy (hapi.im/support). Please upgrade to the latest version to get the best features, bug fixes, and security patches. If you are unable to upgrade at this time, paid support is available for older versions (hapi.im/commercial).
npm WARN deprecated boom@2.[10](https://github.com/HexaInnovLab/Rapidx_Documentation/runs/8091228973?check_suite_focus=true#step:4:11).1: This version has been deprecated in accordance with the hapi support policy (hapi.im/support). Please upgrade to the latest version to get the best features, bug fixes, and security patches. If you are unable to upgrade at this time, paid support is available for older versions (hapi.im/commercial).
npm WARN deprecated hawk@3.1.3: This module moved to @hapi/hawk. Please make sure to switch over as this distribution is no longer supported and may contain bugs and critical security issues.
npm WARN deprecated coffee-script@1.12.7: CoffeeScript on NPM has moved to "coffeescript" (no hyphen)
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated mkdirp@0.3.0: Legacy versions of mkdirp are no longer supported. Please update to mkdirp 1.x. (Note that the API surface has changed to use Promises in 1.x.)
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated har-validator@5.1.5: this library is no longer supported
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
npm WARN deprecated har-validator@4.2.1: this library is no longer supported
npm WARN deprecated request@2.81.0: request has been deprecated, see https://github.com/request/request/issues/3142
npm WARN deprecated svgo@1.3.2: This SVGO version is no longer supported. Upgrade to v2.x.x.
npm WARN deprecated @babel/polyfill@7.12.1: 🚨 This package has been deprecated in favor of separate inclusion of a polyfill and regenerator-runtime (when needed). See the @babel/polyfill docs (https://babeljs.io/docs/en/babel-polyfill) for more information.
npm WARN deprecated puppeteer@2.1.1: Version no longer supported. Upgrade to @latest
npm WARN deprecated svgo@1.3.2: This SVGO version is no longer supported. Upgrade to v2.x.x.
npm WARN deprecated highlight.js@9.18.5: Support has ended for 9.x series. Upgrade to @latest
npm WARN deprecated core-js@2.6.12: core-js@<3.4 is no longer maintained and not recommended for usage due to the number of issues. Because of the V8 engine whims, feature detection in old core-js versions could cause a slowdown up to 100x even if nothing is polyfilled. Please, upgrade your dependencies to the actual version of core-js.

added 2081 packages, and audited 2082 packages in 2m

280 packages are looking for funding
  run `npm fund` for details

75 vulnerabilities (6 low, 30 moderate, 31 high, 8 critical)

To address issues that do not require attention, run:
  npm audit fix

To address all issues possible (including breaking changes), run:
  npm audit fix --force

Some issues need review, and may require choosing
a different dependency.

Run `npm audit` for details.

Running 'npm run build'...


> build
> docusaurus build

[INFO] [en] Creating an optimized production build...
[info] [webpackbar] Compiling Client
[info] [webpackbar] Compiling Server
[success] [webpackbar] Client: Compiled successfully in 43.58s
Imsged[]
cheking []
res1[]
coprights []
coprights []
Imsged[]
cheking []
res1[]
coprights []
coprights []
[success] [webpackbar] Server: Compiled successfully in 48.53s
[WARNING] {"moduleIdentifier":"/github/workspace/website/node_modules/css-loader/dist/cjs.js??ruleSet[1].rules[6].use[1]!/github/workspace/website/node_modules/postcss-loader/dist/cjs.js??ruleSet[1].rules[6].use[2]!/github/workspace/website/node_modules/bootstrap/dist/css/bootstrap.min.css","moduleName":"./node_modules/bootstrap/dist/css/bootstrap.min.css.webpack[javascript/auto]!=!./node_modules/css-loader/dist/cjs.js??ruleSet[1].rules[6].use[1]!./node_modules/postcss-loader/dist/cjs.js??ruleSet[1].rules[6].use[2]!./node_modules/bootstrap/dist/css/bootstrap.min.css","message":"Module Warning (from ./node_modules/postcss-loader/dist/cjs.js):\nWarning\n\n(6:29521) autoprefixer: Replace color-adjust to print-color-adjust. The color-adjust shorthand is currently deprecated.","compilerPath":"client"}
<w> [webpack.cache.PackFileCacheStrategy] Skipped not serializable cache item 'Compilation/modules|/github/workspace/website/node_modules/css-loader/dist/cjs.js??ruleSet[1].rules[6].use[1]!/github/workspace/website/node_modules/postcss-loader/dist/cjs.js??ruleSet[1].rules[6].use[2]!/github/workspace/website/node_modules/bootstrap/dist/css/bootstrap.min.css': No serializer registered for Warning
<w> while serializing webpack/lib/cache/PackFileCacheStrategy.PackContentItems -> webpack/lib/NormalModule -> Array { 1 items } -> webpack/lib/ModuleWarning -> Warning
[SUCCESS] Generated static files in "build".
[INFO] Use `npm run serve` command to test your build locally.
Preparing output...

Copying files to destination directory '/bin/staticsites/ss-oryx/app'...
Done in 10 sec(s).

Removing existing manifest file
Creating a manifest file...
Manifest file created.

Done in 2[11](https://github.com/HexaInnovLab/Rapidx_Documentation/runs/8091228973?check_suite_focus=true#step:4:12) sec(s).


---End of Oryx build logs---
Finished building app with Oryx
[WARNING] Api Directory Location: 'api' could not be found. Azure Functions will not be created.
Either no Api directory was specified, or the specified directory was not found. Azure Functions will not be created.
Zipping App Artifacts
Done Zipping App Artifacts
Uploading build artifacts.
Finished Upload. Polling on deployment.
Status: InProgress. Time: 0.2969081(s)
Status: Failed. Time: [16](https://github.com/HexaInnovLab/Rapidx_Documentation/runs/8091228973?
Deployment Failed :(
Deployment Failure Reason: Deployment Canceled

For further information, please visit the Azure Static Web Apps documentation at https://docs.microsoft.com/en-us/azure/static-web-apps/
If you believe this behavior is unexpected, please raise a GitHub issue at https://github.com/azure/static-web-apps/issues/
Exiting

**Reply by airtower-luna** [reply]
That looks like an Azure-specific problem which I can't comment on. I suspect someone who's familiar with Azure would at least need to see an excerpt of the workflow file.
